src/game/actors/enemy.rs

```diff
@@ -15,41 +15,49 @@ pub struct Enemy {
     torso: i32,
     legs: (i32, i32),
     attack_cooldown: i32,
-    attack_time: i32 // The number of frames in between each attack
+    attack_time: i32, // The number of frames in between each attack
+    map: Dijkstra<'static>,
 }
 
 impl Enemy {
-    pub fn new(x: i32, y: i32, attack_cooldown: i32) -> Enemy {
+    pub fn new(x: i32, y: i32, attack_cooldown: i32, map: Map) -> Enemy {
         return Enemy {
-            x, 
+            x,
             y,
             head: 100,
             arms: (100, 100),
             torso: 100,
             legs: (100, 100),
             attack_cooldown,
-            attack_time: 0
+            attack_time: 0,
+            map: Dijkstra::new_from_map(map, 0_f32),
         };
     }
 
-    pub fn update(&mut self, tiles: &Vec<Box<Tile>>, player: &mut Player) {
+    pub fn update(&mut self, player: &mut Player) {
         // When attack_time reaches attack_cooldown, the enemy can attack
         if self.attack_time < self.attack_cooldown {
             self.attack_time += 1;
-        }
+        } 
         else if player.is_adjacent_to(self) {
             player.calculate_damage(15);
             self.attack_time = 0;
+        } 
+        else if let Some(position) = self.map.walk_one_step() {
+            self.set_position(position);
+            self.attack_time = 0;
         }
     }
 
-    pub fn draw(&self, mut window: &Root) {
-        window.put_char(self.x, self.y, 'E', BackgroundFlag::Set);
-    }
-    
     pub fn clear(&self, mut window: &Root) {
         window.put_char(self.x, self.y, ' ', BackgroundFlag::Set);
     }
+
+    pub fn recalculate_dijkstra(&mut self, destination: (i32, i32)) {
```

Comment:
Per my suggestions for `scene.rs`, this shouldn't be a public function anymore. I can't imagine there's a situation where we want to manually force an enemy to recalculate it's pathing from outside of this file.
```suggestion
    fn recalculate_dijkstra(&mut self, destination: (i32, i32)) {
```

Reply:
Look at commit [709474](https://github.com/team-crogue/crogue/pull/55/commits/7094745bfcfda8f24dcbba34025e5da70432bfb5), this is the solution without outside recalculation. Is this fine?

Reply:
That seems fine, so if you want (later) you can implement that change and commit it so I can re-review.